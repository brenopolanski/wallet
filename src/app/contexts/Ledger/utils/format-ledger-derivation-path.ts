import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
