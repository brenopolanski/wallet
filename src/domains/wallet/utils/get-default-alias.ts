import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
