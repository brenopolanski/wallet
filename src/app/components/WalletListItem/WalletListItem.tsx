import { Contracts } from "@payvo/sdk-profiles";
import { Address } from "app/components/Address";
import { Amount } from "app/components/Amount";
import { Avatar } from "app/components/Avatar";
import { TableCell, TableRow } from "app/components/Table";
import { WalletIcons } from "app/components/WalletIcons";
import { useActiveProfile, useWalletAlias } from "app/hooks";
import cn from "classnames";
import { NetworkIcon } from "domains/network/components/NetworkIcon";
import { isFullySynced } from "domains/wallet/utils/is-fully-synced";
import React, { useMemo } from "react";
import { useTranslation } from "react-i18next";
import { shouldUseDarkColors } from "utils/electron-utils";

export interface WalletListItemProperties {
	wallet: Contracts.IReadWriteWallet;
	activeWalletId?: string;
	onClick?: (walletId: string) => void;
	isCompact?: boolean;
}

export const WalletListItem: React.FC<WalletListItemProperties> = ({
	wallet,
	activeWalletId,
	onClick,
	isCompact = false,
}: WalletListItemProperties) => {
	const { t } = useTranslation();

	const isSelected = useMemo(() => activeWalletId === wallet.id(), [activeWalletId, wallet]);

	const activeProfile = useActiveProfile();

	const canDisplayBalance = isFullySynced(wallet);

	const shadowClasses = useMemo(
		() =>
			cn(
				"group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black",
				{
					"ring-theme-background": !isSelected,
					"ring-theme-success-100": isSelected && !shouldUseDarkColors(),
					"ring-theme-success-900": isSelected && shouldUseDarkColors(),
				},
			),
		[isSelected],
	);

	const { getWalletAlias } = useWalletAlias();

	const { alias } = useMemo(
		() =>
			getWalletAlias({
				address: wallet?.address(),
				network: wallet?.network(),
				profile: activeProfile,
			}),
		[activeProfile, getWalletAlias, wallet],
	);

	let lastCellContent: string | JSX.Element;

	if (wallet.network().isTest() || !canDisplayBalance) {
		lastCellContent = t("COMMON.NOT_AVAILABLE");
	} else {
		lastCellContent = <Amount ticker={wallet.exchangeCurrency()} value={wallet.convertedBalance()} />;
	}

	return (
		<TableRow isSelected={isSelected} onClick={canDisplayBalance ? () => onClick?.(wallet.id()) : undefined}>
			<TableCell variant="start" innerClassName="space-x-4" isCompact={isCompact}>
				<div
					className={cn(
						"flex-shrink-0 flex items-center",
						{ "space-x-4": isCompact },
						{ "-space-x-1": !isCompact },
					)}
				>
					<NetworkIcon
						size="lg"
						network={wallet.network()}
						shadowClassName={shadowClasses}
						isCompact={isCompact}
					/>
					<Avatar size={isCompact ? "xs" : "lg"} address={wallet.address()} shadowClassName={shadowClasses} />
				</div>
				<div className="w-20 flex-1">
					<Address walletName={alias} address={wallet.address()} />
				</div>
			</TableCell>

			<TableCell innerClassName="justify-center text-sm font-bold text-center align-middle" isCompact={isCompact}>
				<div className="inline-flex items-center space-x-1">
					<WalletIcons wallet={wallet} iconSize="lg" />
				</div>
			</TableCell>

			<TableCell innerClassName="font-semibold justify-end" isCompact={isCompact}>
				{canDisplayBalance ? (
					<Amount value={wallet.balance()} ticker={wallet.network().ticker()} />
				) : (
					<span className="text-theme-secondary-text font-normal">{t("COMMON.NOT_AVAILABLE")}</span>
				)}
			</TableCell>

			<TableCell variant="end" innerClassName="justify-end text-theme-secondary-text" isCompact={isCompact}>
				{lastCellContent}
			</TableCell>
		</TableRow>
	);
};
