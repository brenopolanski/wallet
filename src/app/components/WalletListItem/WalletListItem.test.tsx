import userEvent from "@testing-library/user-event";
import { Contracts } from "@payvo/profiles";
import { createMemoryHistory } from "history";
import React from "react";
import { Route } from "react-router-dom";
import { env, fireEvent, getDefaultProfileId, render } from "utils/testing-library";

import { WalletListItem } from "./WalletListItem";

const dashboardURL = `/profiles/${getDefaultProfileId()}/dashboard`;
const history = createMemoryHistory();

let profile: Contracts.IProfile;
let wallet: Contracts.IReadWriteWallet;

describe("WalletListItem", () => {
	beforeAll(() => {
		history.push(dashboardURL);
	});

	beforeEach(async () => {
		profile = env.profiles().findById(getDefaultProfileId());

		wallet = profile.wallets().findById("ac38fe6d-4b67-4ef1-85be-17c5f6841129");
		wallet.data().set(Contracts.WalletFlag.Starred, true);
		wallet.data().set(Contracts.WalletData.DerivationPath, "0");

		await env.profiles().restore(profile);
		await profile.sync();
	});

	it.each([true, false])("should render when isCompact = %s", (isCompact: boolean) => {
		const { container, getByText } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} isCompact={isCompact} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		expect(getByText(wallet.alias()!)).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it("should render for selected wallet", () => {
		jest.spyOn(wallet.network(), "isTest").mockReturnValue(false);

		const walletId = "ac38fe6d-4b67-4ef1-85be-17c5f6841129";

		const { asFragment, queryByText } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} activeWalletId={walletId} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		expect(asFragment()).toMatchSnapshot();
		expect(queryByText("N/A")).toBeNull();
	});

	it("should render with a N/A for fiat", () => {
		jest.spyOn(wallet.network(), "isTest").mockReturnValue(true);

		const walletId = "ac38fe6d-4b67-4ef1-85be-17c5f6841129";

		const { asFragment, getByText } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} activeWalletId={walletId} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		expect(asFragment()).toMatchSnapshot();
		expect(getByText("N/A")).toBeInTheDocument();
	});

	it("should render with default BTC as default exchangeCurrency", () => {
		const mockExchangeCurrency = jest.spyOn(wallet, "exchangeCurrency").mockReturnValue(undefined as any);
		const { container, getByText } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		expect(getByText(wallet.alias()!)).toBeInTheDocument();

		expect(container).toMatchSnapshot();

		mockExchangeCurrency.mockRestore();
	});

	it("should call onClick when clicked and fully restored", () => {
		const onClick = jest.fn();

		const { getByText } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} onClick={onClick} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		userEvent.click(getByText(wallet.alias()!));

		expect(onClick).toHaveBeenCalledWith(wallet.id());
	});

	it("should not call onClick when clicked but not fully restored", () => {
		const hasBeenFullyRestored = jest.spyOn(wallet, "hasBeenFullyRestored").mockReturnValue(false);

		const onClick = jest.fn();

		const { getByText } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} onClick={onClick} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		userEvent.click(getByText(wallet.alias()!));

		expect(onClick).not.toHaveBeenCalled();

		hasBeenFullyRestored.mockRestore();
	});
});
