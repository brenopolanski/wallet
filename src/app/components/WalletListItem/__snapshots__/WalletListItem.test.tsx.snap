// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletListItem should render for selected wallet 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 iMsbqY group"
        data-testid="TableRow"
      >
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 jGsTbn space-x-4"
          >
            <div
              class="flex-shrink-0 flex items-center -space-x-1"
            >
              <div
                aria-label="ARK Devnet"
                class="Circle__CircleWrapper-sc-d5u41q-0 cFnCcb text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700 group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-success-900"
                data-testid="NetworkIcon-ARK-ark.devnet"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  data-testid="NetworkIcon__icon"
                  height="20"
                  width="20"
                >
                  <svg>
                    ark.svg
                  </svg>
                </div>
              </div>
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 fSjBUt group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-success-900 flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cDWumh justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-warning-400"
                data-testid="WalletIcon__Starred"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    star-filled.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cDWumh font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 hNhRDD justify-end text-theme-secondary-text"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              0 BTC
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`WalletListItem should render when isCompact = false 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 gwDXhN group"
        data-testid="TableRow"
      >
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 jGsTbn space-x-4"
          >
            <div
              class="flex-shrink-0 flex items-center -space-x-1"
            >
              <div
                aria-label="ARK Devnet"
                class="Circle__CircleWrapper-sc-d5u41q-0 cFnCcb text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700 group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-background"
                data-testid="NetworkIcon-ARK-ark.devnet"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  data-testid="NetworkIcon__icon"
                  height="20"
                  width="20"
                >
                  <svg>
                    ark.svg
                  </svg>
                </div>
              </div>
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 fSjBUt group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-background flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cDWumh justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-warning-400"
                data-testid="WalletIcon__Starred"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    star-filled.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__TestNetwork"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    code.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cDWumh font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 hNhRDD justify-end text-theme-secondary-text"
          >
            N/A
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WalletListItem should render when isCompact = true 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 gwDXhN group"
        data-testid="TableRow"
      >
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 jtRNyC space-x-4"
          >
            <div
              class="flex-shrink-0 flex items-center space-x-4"
            >
              <div
                aria-label="ARK Devnet"
                class="w-5 h-5 inline-flex items-center text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700"
                data-testid="NetworkIcon-ARK-ark.devnet"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  data-testid="NetworkIcon__icon"
                  height="20"
                  width="20"
                >
                  <svg>
                    ark.svg
                  </svg>
                </div>
              </div>
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 hiKOvx group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-background flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 dEQQQI justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-warning-400"
                data-testid="WalletIcon__Starred"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    star-filled.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__TestNetwork"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    code.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 dEQQQI font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 kXCVVq justify-end text-theme-secondary-text"
          >
            N/A
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WalletListItem should render with a N/A for fiat 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 iMsbqY group"
        data-testid="TableRow"
      >
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 jGsTbn space-x-4"
          >
            <div
              class="flex-shrink-0 flex items-center -space-x-1"
            >
              <div
                aria-label="ARK Devnet"
                class="Circle__CircleWrapper-sc-d5u41q-0 cFnCcb text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700 group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-success-900"
                data-testid="NetworkIcon-ARK-ark.devnet"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  data-testid="NetworkIcon__icon"
                  height="20"
                  width="20"
                >
                  <svg>
                    ark.svg
                  </svg>
                </div>
              </div>
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 fSjBUt group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-success-900 flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cDWumh justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-warning-400"
                data-testid="WalletIcon__Starred"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    star-filled.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__TestNetwork"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    code.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cDWumh font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 hNhRDD justify-end text-theme-secondary-text"
          >
            N/A
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`WalletListItem should render with default BTC as default exchangeCurrency 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 gwDXhN group"
        data-testid="TableRow"
      >
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 jGsTbn space-x-4"
          >
            <div
              class="flex-shrink-0 flex items-center -space-x-1"
            >
              <div
                aria-label="ARK Devnet"
                class="Circle__CircleWrapper-sc-d5u41q-0 cFnCcb text-theme-secondary-700 border-theme-secondary-300 dark:border-theme-secondary-700 group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-background"
                data-testid="NetworkIcon-ARK-ark.devnet"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  data-testid="NetworkIcon__icon"
                  height="20"
                  width="20"
                >
                  <svg>
                    ark.svg
                  </svg>
                </div>
              </div>
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 fSjBUt group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black ring-theme-background flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cDWumh justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-warning-400"
                data-testid="WalletIcon__Starred"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    star-filled.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__TestNetwork"
              >
                <div
                  class="sc-bdnxRM NjstW"
                  height="20"
                  width="20"
                >
                  <svg>
                    code.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cDWumh font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 hNhRDD justify-end text-theme-secondary-text"
          >
            N/A
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
